src/test/scala/mesosphere/marathon/SchedulerDriverFactoryTest.scala

```diff
@@ -0,0 +1,52 @@
+package mesosphere.marathon
+
+import mesosphere.AkkaUnitTest
+import mesosphere.marathon.core.base.CrashStrategy
+import mesosphere.marathon.core.instance.{Instance, TestInstanceBuilder}
+import mesosphere.marathon.core.storage.store.impl.memory.InMemoryPersistenceStore
+import mesosphere.marathon.metrics.dummy.DummyMetrics
+import mesosphere.marathon.state.PathId._
+import mesosphere.marathon.storage.repository.{FrameworkIdRepository, InstanceRepository}
+import mesosphere.marathon.test.TestCrashStrategy
+
+import org.scalatest.Inside
+import scala.util.{Try, Failure}
+
+class SchedulerDriverFactoryTest extends AkkaUnitTest with Inside {
+  val metrics = DummyMetrics
+  val appId = "/test/foo/bla/rest".toPath
+  val instanceId = Instance.Id.forRunSpec(appId)
+
+  class Fixture {
+    val store = new InMemoryPersistenceStore(metrics)
+    val zkTimeout = implicitly[PatienceConfig].timeout
+    val frameworkIdRepository = FrameworkIdRepository.inMemRepository(store)
+    val instanceRepository = InstanceRepository.inMemRepository(store)
+    val crashStrategy = new TestCrashStrategy()
+    store.markOpen()
+  }
+
+  "getFrameworkId throws an exception and crashes if frameworkId is undefined but there are instances defined" in new Fixture {
+    instanceRepository.store(TestInstanceBuilder.emptyInstance(instanceId = instanceId))
```

Comment:
The test fails on jenkins with
```
The partial function passed as the second parameter to inside was not defined
at the value passed as the first parameter to inside, which was: Success(None)
(SchedulerDriverFactoryTest.scala:37)
```

because it doesn't wait for the instance to be stored. Append `.futureValue`?

Reply:
Danke :) stupid oversight on my part.

Reply:
- [ ] I'll add the `.futureValue`